src/main/java/com/example/BookService/Service/BookService.java

```diff
@@ -0,0 +1,84 @@
+package com.example.BookService.Service;
+
+import com.example.BookService.DTO.BookDTO;
+import com.example.BookService.Entity.Book;
+import com.example.BookService.Repository.BookRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class BookService {
+
+    private final BookRepository bookRepository;
+    private final WebClient webClient;
+    public BookService(BookRepository bookRepository, WebClient.Builder webClientBuilder) {
+        this.bookRepository = bookRepository;
+        this.webClient=webClientBuilder.baseUrl("http://order-service:8080").build();
```

Comment:
You can extract the baseurl as a constant so we make sure there is just 1 place where we have it.